openml/runs/functions.py

```diff
@@ -1094,6 +1094,19 @@ def list_runs(offset=None, size=None, id=None, task=None, setup=None,
         List of found runs.
     """
 
+    if id is not None and (not type(id) is list):
+        raise Exception('id must be of type list.')
```

Comment:
It is good practice to raise a  [`TypeError`](https://docs.python.org/3.6/library/exceptions.html#TypeError) when the passed argument is not of the correct type. This makes it easier for the caller to check against certain types of errors, and the nature of the error is more immediately clear.

Reply:
Made that change